operators/advanced_network/advanced_network/managers/dpdk/adv_network_dpdk_mgr.cpp

```diff
@@ -1958,7 +1934,10 @@ void DpdkMgr::free_tx_metadata(BurstParams* burst) {
 
 Status DpdkMgr::get_tx_metadata_buffer(BurstParams** burst) {
   if (rte_mempool_get(tx_metadata, reinterpret_cast<void**>(burst)) != 0) {
-    HOLOSCAN_LOG_CRITICAL("Failed to get TX meta descriptor");
+    HOLOSCAN_LOG_CRITICAL("Running out of TX meta buffers due to high rates. Either increase "\
+      "your number of metadata buffers (current: {}) with `tx_meta_buffers` (will "\
+      "increase memory usage) or increase your `batch_size` for port {} queue {} (will increase "\
+      "latency)", cfg_.tx_meta_buffers_, (*burst)->hdr.hdr.port_id, (*burst)->hdr.hdr.q_id);
```

Comment:
Should that message be common to all managers? could we have that log in a base implementation when a manager-specific implementation returns NO_FREE_BURST_BUFFERS ? Ok to keep as improvement for follow-up if you agree, we can just track

Reply:
This goes back to the discussion about if all the backends should use the same buffer pools and rings. If we do that then this function would move into the manager.cpp file

Reply:
Cool, let me create a ticket for that.